src/Compilers/CSharp/Test/Semantic/Semantics/SimpleProgramsTests.cs

```diff
@@ -0,0 +1,1837 @@
+
+            // PROTOTYPE(SimplePrograms): All errors should be ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement
+            comp.VerifyDiagnostics(
+                // (13,34): error CS0119: 'Test' is a type, which is not valid in the given context
+                //         System.Console.WriteLine(Test); // 1
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "Test").WithArguments("Test", "type").WithLocation(13, 34),
+                // (14,9): error CS0120: An object reference is required for the non-static field, method, or property 'object.ToString()'
+                //         Test.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "Test.ToString").WithArguments("object.ToString()").WithLocation(14, 9),
+                // (15,14): error CS0117: 'Test' does not contain a definition for 'EndsWith'
+                //         Test.EndsWith(null); // 3
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "EndsWith").WithArguments("Test", "EndsWith").WithLocation(15, 14),
+                // (31,38): error CS0119: 'Test' is a type, which is not valid in the given context
+                //             System.Console.WriteLine(Test); // 5
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "Test").WithArguments("Test", "type").WithLocation(31, 38),
+                // (32,13): error CS0120: An object reference is required for the non-static field, method, or property 'object.ToString()'
+                //             Test.ToString(); // 6
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "Test.ToString").WithArguments("object.ToString()").WithLocation(32, 13),
+                // (33,18): error CS0117: 'Test' does not contain a definition for 'EndsWith'
+                //             Test.EndsWith(null); // 7
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "EndsWith").WithArguments("Test", "EndsWith").WithLocation(33, 18)
+                );
+        }
+
+        [Fact]
+        public void Scope_03()
+        {
+            var text1 = @"
+string Test = ""1"";
+System.Console.WriteLine(Test);
+";
+            var text2 = @"
+class Test {}
+
+class Derived : Test
+{
+    void M()
+    {
+        int Test = 0;
+        System.Console.WriteLine(Test++);
+    }
+}
+
+namespace N1
+{
+    class Derived : Test
+    {
+        void M()
+        {
+            int Test = 1;
+            System.Console.WriteLine(Test++);
+        }
+    }
+}
+";
+
+            var comp = CreateCompilation(new[] { text1, text2 }, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            comp.VerifyDiagnostics();
+
+            comp = CreateCompilation(text1 + text2, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            comp.VerifyDiagnostics();
+
+            Assert.Throws<System.ArgumentException>(() => CreateCompilation(new[] { Parse(text1, filename: "text1", DefaultParseOptions),
```

Comment:
Why does this throw? Because of inconsistent parse options between syntax trees?

Reply:
> Why does this throw? Because of inconsistent parse options between syntax trees?

Yes. This PR doesn't change this behavior.

---
In reply to: [382937604](https://github.com/dotnet/roslyn/pull/41706#discussion_r382937604) [](ancestors = 382937604)